opm/material/fluidsystems/blackoilpvt/GasPvtThermal.hpp

```diff
@@ -392,6 +396,10 @@ class GasPvtThermal
     const Scalar gasReferenceDensity(unsigned regionIdx) const
     { return isothermalPvt_->gasReferenceDensity(regionIdx); }
 
+    const Scalar hVap(unsigned regionIdx) const {
+        return this->hVap_[regionIdx];
+    }
```

Comment:
How will you compute/set hVAP?

Reply:
We have a separate pullrequest on this. HEATVAP. The other part is that of the pullrequest is to make a mixing model for blackoil similar to what you have done in co2store. The trouble is that the energy of one phase depend on the properties of the other component so this is done in Blackoilfluid system.

For may gases + liquids internal energy depend only on temperature, while enthalpy depend on pressure so it is easiest to make functions for internal energy. Also it is usefull to drop the work term i.e. PV term in transport in some calculations since the behavoir is simpler.

The main motivation is to make a reasonable model for blackoil.